<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>630</width>
    <height>632</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>611</width>
      <height>611</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>lab1 - lab2</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>330</y>
        <width>351</width>
        <height>81</height>
       </rect>
      </property>
      <property name="title">
       <string>lab2</string>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>331</width>
         <height>29</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QLabel" name="label_lab2">
          <property name="text">
           <string>input</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_inptG_lab2"/>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_go_lab2">
          <property name="text">
           <string>run</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>50</y>
        <width>351</width>
        <height>271</height>
       </rect>
      </property>
      <property name="title">
       <string>lab1</string>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>331</width>
         <height>231</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <item>
           <widget class="QPushButton" name="pushButton_run">
            <property name="text">
             <string>RUN</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_clear_2">
            <property name="text">
             <string>CLEAR</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="textEdit_result">
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>581</width>
        <height>35</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>22</pointsize>
       </font>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label_first">
         <property name="font">
          <font>
           <family>Times New Roman</family>
           <pointsize>22</pointsize>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>G = ({</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_VT">
         <property name="font">
          <font>
           <family>Times New Roman</family>
           <pointsize>22</pointsize>
          </font>
         </property>
         <property name="text">
          <string>VT</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_mid">
         <property name="font">
          <font>
           <family>Times New Roman</family>
           <pointsize>22</pointsize>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>}, {</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_VN">
         <property name="font">
          <font>
           <family>Times New Roman</family>
           <pointsize>22</pointsize>
          </font>
         </property>
         <property name="text">
          <string>VN</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_last">
         <property name="font">
          <font>
           <family>Times New Roman</family>
           <pointsize>22</pointsize>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>}, P, S)</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>220</width>
        <height>111</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>VN</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_VN">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="maxLength">
            <number>20</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>VT</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_VT">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="maxLength">
            <number>20</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_reload">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Reload</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>250</y>
        <width>221</width>
        <height>321</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLineEdit" name="lineEdit_P">
           <property name="maxLength">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>-&gt;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_Rule">
           <property name="maxLength">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_Add">
           <property name="text">
            <string>Add</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTextEdit" name="textEdit_AllRule">
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_clear">
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>221</width>
        <height>71</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Max Len</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_max">
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="maximum">
            <number>20</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButton_rigth">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Rigth</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Min Len </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_min">
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="maximum">
            <number>20</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButton_left">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Left  </string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>lab 3</string>
     </attribute>
     <widget class="QWidget" name="">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>10</y>
        <width>361</width>
        <height>251</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_18">
         <item>
          <widget class="QLabel" name="label_12">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>(</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_position">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_13">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>,</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_index">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_14">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>,</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_stack">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_15">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>) -&gt;(</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_position_2">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_16">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>,</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_stack_2">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_17">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_addB">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Add</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTextEdit" name="textEdit_B">
         <property name="font">
          <font>
           <family>Times New Roman</family>
           <pointsize>14</pointsize>
          </font>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_19">
         <item>
          <spacer name="horizontalSpacer_5">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_clearB">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>214</width>
        <height>251</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_11">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Q</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_Q">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_12">
         <item>
          <widget class="QLabel" name="label_7">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>V</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_V">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_13">
         <item>
          <widget class="QLabel" name="label_8">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Z </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_Z">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_14">
         <item>
          <widget class="QLabel" name="label_9">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>q0</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_q0">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_15">
         <item>
          <widget class="QLabel" name="label_10">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Z0</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_Z0">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_16">
         <item>
          <widget class="QLabel" name="label_11">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>F  </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_F">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>14</pointsize>
            </font>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_17">
         <item>
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_R">
           <property name="text">
            <string>Reload</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
